project/build.scala

```diff
@@ -1,6 +1,147 @@
-package org.scalamacros
+import bintray.BintrayPlugin.autoImport._
+import ch.epfl.scala.sbt.release.ReleaseEarlyPlugin.autoImport._
+import com.typesafe.sbt.SbtPgp.autoImport._
+import dotty.tools.sbtplugin.DottyPlugin.autoImport._
+import sbt.Def
+import sbt.Keys._
+import sbt._
+import sbt.plugins.JvmPlugin
+import sbtassembly.AssemblyPlugin.autoImport._
+import sbtbuildinfo.BuildInfoPlugin
+import sbtbuildinfo.BuildInfoPlugin.autoImport._
 
-import org.scalamacros.build._
+object Version {
+  val scala212 = "2.12.3"
+  val scala213 = "2.13.0-M1"
+  val dotty = "0.3.0-RC2"
+}
 
-trait ScalamacrosBuild extends Products with Versions
-package object build extends ScalamacrosBuild
+object ScalamacrosBuildPlugin extends AutoPlugin {
+  import Version._
+  override def requires: Plugins = JvmPlugin && BuildInfoPlugin
+  override def trigger: PluginTrigger = allRequirements
+
+  object autoImport {
+    lazy val pluginSettings = Def.settings(
+      crossVersion := CrossVersion.full,
+      unmanagedSourceDirectories.in(Compile) += {
+        val base = sourceDirectory.in(Compile).value
+        base / ("scala-" + scalaVersion.value)
+      },
+      test.in(assembly) := {},
+      logLevel.in(assembly) := Level.Error,
+      assemblyJarName.in(assembly) := {
+        name.value + "_" + scalaVersion.value + "-" + version.value + "-assembly.jar"
+      },
+      assemblyOption.in(assembly) ~= { _.copy(includeScala = false) },
+      Keys.`package`.in(Compile) := {
+        val slimJar = Keys.`package`.in(Compile).value
+        val fatJar = new File(crossTarget.value + "/" + assemblyJarName.in(assembly).value)
+        val _ = assembly.value
+        IO.copy(List(fatJar -> slimJar), overwrite = true)
+        slimJar
```

Comment:
If the artifacts will be generated with sbt-assembly shouldn't dependencies like [shaded](https://github.com/sbt/sbt-assembly#shading)?

Reply:
This artifact should only be used via `addCompilerPlugin`/`-Xplugin:path/to/macros.jar`. Compiler plugins are loaded in an isolated classloader from other dependencies.